### ISSUE TITLE
[tracking] shim 15.4 critical regressions

### ISSUE THREAD

**Issue by julian-klode** [rhboot/shim-review#165]
Posting this here for awareness of submitters regarding fatal issues that can break boot on machines and other errors of unknown extend.

Critical issues:

* fails to boot on older Macs, and other machines with EFI < 2: https://github.com/rhboot/shim/pull/364
* mokutil --disable-validation does not work: https://github.com/rhboot/shim/pull/362
* 32-bit Intel is broken: https://github.com/rhboot/shim/pull/357
* various systems, especially older ones that do not support secure boot, fail to mirror MOK variables. rhboot/shim#372 makes that non-fatal for non-secure systems which is at least something
* fwupd and similar cannot be loaded, load option parsing is broken. Patches fixing that in https://github.com/rhboot/shim/pull/379

Important issues:
* 64-bit ARM is broken on old binutils; unreproducible with new ones: https://github.com/rhboot/shim/issues/366

Unknown extent:

* produces kernel errors: https://github.com/rhboot/shim/pull/361

You might want to hold off submitting shims until there is a 15.5 release, or cherry-pick patches.

**Comment by Doncuppjr** [comment]
So we need 822d07ad4f07ef66fe447a130e1027c88d02a394 if we want Mac support?

**Comment by steve-mcintyre** [comment]
aarch64 builds are really awkward - they will break using binutils versions before 2.35, and even after that they may give unpredictable results. We really need proper toolchain support.

**Comment by Doncuppjr** [comment]
I guess the better question is, do they want to make a 15.5?

**Comment by steve-mcintyre** [comment]
822d07ad4f07ef66fe447a130e1027c88d02a394 is for really old Macs that won't do SB anyway. But without that patch you'll fail to boot the machine at all

**Comment by Doncuppjr** [comment]
And I guess we are waiting on 8b59591775a0412863aab9596ab87bdd493a9c1e to be merged or QA'ed?

**Comment by cmurf** [comment]
Fedora's shim-15.4-4 has fixes for some of the items listed in this issue. But there are two issues we're seeing with a subset of Lenovo and Dell systems that don't yet have upstream issues filed (and I don't understand the problem well enough to file them).
https://bugzilla.redhat.com/show_bug.cgi?id=1955416
https://bugzilla.redhat.com/show_bug.cgi?id=1954245

**Comment by julian-klode** [comment]
shim 15.5 is out, closing this.
